### ISSUE TITLE
[licensing] Don't cache (Replicated) license but call the kots server for each license call

### ISSUE THREAD

**Issue by corneliusludmann** [gitpod-io/gitpod#8414]
Currently, the license is fetched from the KOTS admin server on server start. When the license has been changed, we need to restart the server.

I would propose to call the KOTS license endpoint for each license check instead of holding the license in the cache.

Drawback:
- More API calls.

However, we do not check the license that often in Gitpod. And the KOTS endpoint is in the same network. That's why I think that would be ok.

Alternative: Refetch the license regularly (e.g. once an hour or day).

**Comment by MrSimonEmms** [comment]
I don't think this is needed.

KOTS automatically syncs the license between the `kotsadm` pod and the Replicated back-end (I think it's every 4 hours) and then prompts the user to redeploy the application with the new licence if it detects a change.

Gitpod also only checks the licence when `server` starts up, making an HTTP call to `http://kotsadmn:3000/license/v1/license`. This API is never cached, meaning that if the call is made again, it will get the latest licence data (regardless of whether we've redeployed the licence via the KOTS admin page). The licence is then stored in memory.

To my mind, I would suggest keeping it as-is and letting KOTS do the heavy lifting.

**Comment by corneliusludmann** [comment]
> KOTS automatically syncs the license between the `kotsadm` pod and the Replicated back-end (I think it's every 4 hours) and then prompts the user to redeploy the application with the new licence if it detects a change.

That's true. But does a redeploy always mean that the server restarts? A redeploy is a `gitpod-installer render … | kubectl apply` but when the server manifest has no changes (because it's just a license change and no Installer change) will the pod restart anyways?

**Comment by MrSimonEmms** [comment]
A very good question indeed. In my testing, I've found that it does tend to restart, but there's nothing in the code that would make it intrinsically do so.

I'll need to investigate this further

**Comment by lucasvaltl** [comment]
👋 I've tested this just now on my self-hosted instance. I can confirm that this _should_ not be an issue:
1. I created a license for 8 users, and used it during install of my instance.
2. Went ahead and completed the installation - validated in the admin dashboard > License tab that the license is for 8 users
3. Changed the license in the replicated vendor portal to be for 12 users
4. Synced license in the Kots Admin UI, and then waited for and redeployed with said license (I did redeploy here, which is expected on a license change)
5. Went into the admin dashboard > License tab again and validated that the license now says for 12 users.
--> As such, this should be a non issue. If you agree @corneliusludmann, we can close this issue.
